File: language/compiler/src/com/kubadziworski/parsing/visitor/CompilationUnitVisitor.java
package com.kubadziworski.parsing.visitor;

import jiva.parser.JivaBaseVisitor;
import jiva.parser.JivaParser;
import jiva.parser.JivaParser.ClassDeclarationContext;
import jiva.parser.JivaParser.CompilationUnitContext;
import com.kubadziworski.domain.CompilationUnit;
import com.kubadziworski.domain.ClassDeclaration;
import org.antlr.v4.runtime.misc.NotNull;

/**
 * Created by kuba on 28.03.16.
 */
public class CompilationUnitVisitor extends JivaBaseVisitor<CompilationUnit> {

    @Override
    public CompilationUnit visitCompilationUnit(@NotNull CompilationUnitContext ctx) {
        ClassVisitor classVisitor = new ClassVisitor();
        ClassDeclarationContext classDeclarationContext = ctx.classDeclaration();
        ClassDeclaration classDeclaration = classDeclarationContext.accept(classVisitor);
        return new CompilationUnit(classDeclaration);
    }
}

File: language/compiler/src/parsing/visitor/CompilationUnitVisitor.kt
package jiva.parsing.visitor

import jiva.domain.CompilationUnit
import jiva.parser.JivaBaseVisitor
import jiva.parser.JivaParser.CompilationUnitContext
import org.antlr.v4.runtime.misc.NotNull

/**
 * Created by kuba on 28.03.16.
 */
class CompilationUnitVisitor : JivaBaseVisitor<CompilationUnit?>() {
    
    override fun visitCompilationUnit(ctx: CompilationUnitContext): CompilationUnit {
        val classVisitor = ClassVisitor()
        val classDeclarationContext = ctx.classDeclaration()
        val classDeclaration = classDeclarationContext.accept(classVisitor)!!
        return CompilationUnit(classDeclaration)
    }
    
}
